manual/trees/index.md

```diff
@@ -7,178 +7,187 @@ toc: true
+> **Note**
+>
+> Do *not* use the {% include ref class="TBranch" %} constructor to add a branch to a tree.
 
-The `cernbuild.C` ROOT macro creates a root file (`cernstaff.root`) and prints the tree `T` and its branches with [TTree::Print()](https://root.cern/doc/master/classTTree.html#a7a0006d38d5066b533e040aa16f97094){:target="_blank"}.
+> **Note**
+>
+> The objects / variables used to create branches must not be destroyed until the `TTree` is deleted or `TTree::ResetBranchAddress()` is called.
+
+**1. Branches holding basic types**
+
+If you have a variable of type `int`, `float`, `bool`, or any other basic type, you can create a branch (and a leaf) from it.
+For fundamental datatypes, the type can be deduced from the variable and the name of the leaf will be set to the name of the branch.
+In Python, that type information is not available and the leaf name and data type must be specified as third argument.
+Further details are explained in the [reference guide](https://root.cern.ch/doc/master/classTTree.html#addcolumnoffundamentaltypes).
 
 {% highlight C++ %}
-root [0] .x cernbuild.C
-******************************************************************************
-*Tree    :T         : CERN 1988 staff data                                   *
-*Entries :     3354 : Total =          176339 bytes  File  Size =      15005 *
-*        :          : Tree compression factor =   2.74                       *
-******************************************************************************
-*Br    0 :Category  : Category/I                                             *
-*Entries :     3354 : Total  Size=      14073 bytes  One basket in memory    *
-*Baskets :        0 : Basket Size=      32000 bytes  Compression=   1.00     *
-*............................................................................*
-*Br    1 :Flag      : Flag/i                                                 *
-*Entries :     3354 : Total  Size=      14049 bytes  One basket in memory    *
-*Baskets :        0 : Basket Size=      32000 bytes  Compression=   1.00     *
-*............................................................................*
-*Br    2 :Age       : Age/I                                                  *
-*Entries :     3354 : Total  Size=      14043 bytes  One basket in memory    *
-*Baskets :        0 : Basket Size=      32000 bytes  Compression=   1.00     *
-*............................................................................*
-*Br    3 :Service   : Service/I                                              *
-*Entries :     3354 : Total  Size=      14067 bytes  One basket in memory    *
-*Baskets :        0 : Basket Size=      32000 bytes  Compression=   1.00     *
-*............................................................................*
-...
-...
-...
+float var;
+tree->Branch("branch0", &var);
+{% endhighlight %}
+
+{% highlight Python %}
+# Provide a one-element array, so ROOT can read data from this memory. 
```

Comment:
Maybe clarify a bit this comment.

Reply:
Discusses with Enric, clarification can quickly go out of control here :-) and we agree this is "good enoough".